### DISCUSSION TITLE
No transmitter solution

### DISCUSSION THREAD

**Discussion by kobelev-tech** [headtracker/HeadTracker#88] [Q&A]
Dear dlktdr,

Your documentation mentions "Option 3 - Bridged". In this case a second Nano 33 BLE will send out SBUS or PPE signal that can be fed into an RC transmitter or maybe into some other device.

I would like to discuss somewhat similar, although a different setup. I want to attach servos directly to this second Nano 33 BLE. I understand that your project cannot be used for this as-is. How do you think, will that require complex modifications and if this is possible at all? A couple of questions:

1. Bluetooth specs say that its distance is about 10m. Have you tried your particular devices at that extreme distance? Would it be reasonable to expect that this will work reliably when the distance is not more than 5m and there are 1-2 drywalls in between?

2. Do you think that it would be enough just to put +4.8v (or whatever it needs) on the servo as power and simply connect pin from Arduino to the PWM input of the servo? Can it be that some other version of hardware (not Nano 33 BLE) will make more sense here?

3. How complex will be the changes on the software side? Can you give estimate in men-weeks? Will you be interested in participating in development of this change or for example help me? I have good mainframe/microcontroller/software background but never did projects for Arduino.

Reasonable donation can be discussed.

Thanks,
Kirill

**Comment by dlktdr** [comment]
1) Bluetooth for me has been limited on the range. with the amount of data being sent I have been able to get at most 3-5m on the best day with no other interference around. As you get further away some packets are dropped causing jitter in the servos. Some users who have an alternate Bluetooth module in their radio's can't get more than 1 - 1.5m away.

2) You can currently use the code as is on the receiver or transmitter directly with up to four servos on pins A0-A4. Because the signal is only 3.3V output I have had some servo's respond just fine while others will not. You need a separate 5V supply for their power still. https://headtracker.gitbook.io/head-tracker/settings/gui-settings/pwm

3) I can certainly point you to where you would need to make the change in the code to get more than 4ch. Your other option is to pick up a SBUS to PWM converter, you could use all 8 Bluetooth channels with no changes this way and the voltage levels will be correct. Might be the best option. https://www.ebay.ca/itm/184954631107 https://www.ebay.ca/itm/353844306111

**Reply by kobelev-tech** [reply]
I see that this is not an option. Unfortunately 3m guaranteed range is too little. No problem. We have what we have.

Let me ask another Q: What would be your recommendation for a transmitter brand in the price range about 500 USD?
The best if it will be also compatible with Spectrum DSM2/DSMX receivers.

Thanks again.

**Comment by dlktdr** [comment]
I have an update. I just read about PHY coded mode for BLE and was able to integrate it into the code. Haven't done a real range test yet but from what I have read should drastically increase the distance. So far 5m across my room is no problem nice and smooth output.

If you are looking for a radio I would suggest the RM TX16S Mk2 with a 4in1 module. It's a nice radio, works with many receivers including Spektrum DSM2/DSMX. There are others like by radiomaster with a 4 in 1 module too. Can be had for much less than 500usd.
